coltypes/coltypes.go

```diff
@@ -110,3 +129,100 @@ func GetColumnTypes(ctx context.Context, repo *repository.Oracle, tableName stri
 
 	return columnTypes, nil
 }
+
+// FormatData formats the data into Oracle types
+// and wraps the placeholder in a special function, if necessary.
+func FormatData(
+	columnTypes map[string]ColumnData,
+	key string, value any,
+) (string, any, error) {
+	if value == nil {
+		return QueryPlaceholder, nil, nil
+	}
+
+	switch oracleColumnData := columnTypes[strings.ToUpper(key)]; {
+	// if the column type is a NUMBER, precision is 1, scale is 0, and value is a boolean,
+	// convert it to the integer, when the "true" value is 1, and "false" is 0
+	case oracleColumnData.Type == oracleTypeNumber &&
+		oracleColumnData.Precision != nil && *oracleColumnData.Precision == 1 &&
```

Comment:
Do you mean move the next checks to another function?
```
oracleColumnData.Precision != nil && *oracleColumnData.Precision == 1 &&
		oracleColumnData.Scale != nil && *oracleColumnData.Scale == 0
```

Because, in this case, we do not have to check all NUMBER types, just those which have Precision = 1 and Scale = 0 data.